fatal unhandled exception for type error where sequence is used as index

This ought to be a simple type error, but it gives a fatal unhandled exception instead.

```
method f(a: seq<int>) {
  var t := a[a .. 1];
}
```

```
$ dafny a.dfy 
Dafny 2.3.0.10506
[ERROR] FATAL UNHANDLED EXCEPTION: System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
  at System.Text.StringBuilder.AppendFormatHelper (System.IFormatProvider provider, System.String format, System.ParamsArray args) [0x000f9] in <b4d1a66727f44ba888b54966284fabb4>:0 
  at System.String.FormatHelper (System.IFormatProvider provider, System.String format, System.ParamsArray args) [0x00023] in <b4d1a66727f44ba888b54966284fabb4>:0 
  at System.String.Format (System.String format, System.Object[] args) [0x00020] in <b4d1a66727f44ba888b54966284fabb4>:0 
  at Microsoft.Dafny.ErrorReporter.Error (Microsoft.Dafny.MessageSource source, Microsoft.Boogie.IToken tok, System.String msg, System.Object[] args) [0x00001] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver+TypeConstraint+ErrorMsg.Reporting (Microsoft.Dafny.ErrorReporter reporter, System.String suffix) [0x0002e] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver+TypeConstraint+ErrorMsg.ReportAsError (Microsoft.Dafny.ErrorReporter reporter) [0x00012] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver+TypeConstraint.ReportErrors (Microsoft.Dafny.ErrorReporter reporter) [0x0001b] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver.SolveAllTypeConstraints () [0x000f9] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver.ResolveMethod (Microsoft.Dafny.Method m) [0x006bf] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver.ResolveClassMemberBodies (Microsoft.Dafny.ClassDecl cl) [0x0019e] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver.ResolveTopLevelDecls_Core (System.Collections.Generic.List`1[T] declarations, Microsoft.Dafny.Graph`1[Node] datatypeDependencies, Microsoft.Dafny.Graph`1[Node] codatatypeDependencies) [0x008f8] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver.ResolveModuleDefinition (Microsoft.Dafny.ModuleDefinition m, Microsoft.Dafny.ModuleSignature sig) [0x00277] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Resolver.ResolveProgram (Microsoft.Dafny.Program prog) [0x004e6] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Main.Resolve (Microsoft.Dafny.Program program, Microsoft.Dafny.ErrorReporter reporter) [0x00036] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.Main.ParseCheck (System.Collections.Generic.IList`1[T] files, System.String programName, Microsoft.Dafny.ErrorReporter reporter, Microsoft.Dafny.Program& program) [0x0001e] in <b8a04a8c5d6040ac97a2fa1d622f53e6>:0 
  at Microsoft.Dafny.DafnyDriver.ProcessFiles (System.Collections.Generic.IList`1[T] dafnyFiles, System.Collections.ObjectModel.ReadOnlyCollection`1[T] otherFileNames, Microsoft.Dafny.ErrorReporter reporter, System.Boolean lookForSnapshots, System.String programId) [0x001d8] in <faa1593fe210425eaf9f4a17ce9317fe>:0 
  at Microsoft.Dafny.DafnyDriver.ThreadMain (System.String[] args) [0x0003d] in <faa1593fe210425eaf9f4a17ce9317fe>:0 
  at Microsoft.Dafny.DafnyDriver+<>c__DisplayClass1_0.<Main>b__0 () [0x00001] in <faa1593fe210425eaf9f4a17ce9317fe>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <b4d1a66727f44ba888b54966284fabb4>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <b4d1a66727f44ba888b54966284fabb4>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <b4d1a66727f44ba888b54966284fabb4>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <b4d1a66727f44ba888b54966284fabb4>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <b4d1a66727f44ba888b54966284fabb4>:0 
```

--- a/Test/git-issues/git-issue-343.dfy
+++ b/Test/git-issues/git-issue-343.dfy
@@ -1,0 +1,6 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+method f(a: seq<int>) {
+  var t := a[a .. 1];
+}

--- a/Test/git-issues/git-issue-343.dfy.expect
+++ b/Test/git-issues/git-issue-343.dfy.expect
@@ -1,0 +1,3 @@
+git-issue-343.dfy(5,13): Error: wrong number of indices for multi-selection
+git-issue-343.dfy(5,13): Error: incorrect type for selection into seq<int> (got seq<int>)
+2 resolution/type errors detected in git-issue-343.dfy

